build_tools/ci/cpu_comparison/input_generator.py

```diff
@@ -0,0 +1,151 @@
+# This python script is expected to be run from the command-line with 2
+# arguments:
+#
+#   1) the name of a file to parse.
+#   2) the directory where binary files will be written.
+#
+# Example:
+# ```
+# python input_generator.py <input_file> <output_dir>
+# ```
+#
+# The file <input_file> contains an mlir function, and header information
+# about the inputs. See existing tests for examples.
+#
+# The header information about inputs is lines containing input types.
+# Example:
+#
+# ```
+#  # input 3x40xf32
+#  # input 2x2xi32
+# ```
+# This script finds all lines of the form above and generates binary files with
+# random data for them.
+#
+# This script also create a file containing a single line of the form
+# `--input="3x40xf32=@<binary_file>" --input="2x2xi32=@<binary_file>"`
+#
+# which will be used as input to iree-run-module in the main script.
+
+
+import numpy as np
+import struct
+import sys
+import os
+
+
+def float32_to_bfloat16(float32_value):
+
+    # Convert to a 32-bit integer representation
+    int32_repr = float32_value.view(np.int32)
+
+    # Truncate the lower 16 bits
+    bf16_int_repr = (int32_repr >> 16) & 0xFFFF
+
+    # Convert to numpy uint16 for consistency
+    return np.uint16(bf16_int_repr)
+
+
+def generate_bfloat16_data(num_values, lower_bound, upper_bound):
+
+    float_data = np.random.randint(lower_bound, upper_bound, num_values).astype(
+        np.float32
+    )
+
+    # Convert float32 data to bfloat16
+    bf16_data = [float32_to_bfloat16(f) for f in float_data]
+
+    # Pack bfloat16 data into binary format
+    binary_data = struct.pack(f"{len(bf16_data)}H", *bf16_data)
+
+    return binary_data
+
+
+def get_numpy_type(element_type):
+    if element_type == "float32" or element_type == "f32":
+        return np.float32
+    elif element_type == "int32" or element_type == "i32":
+        return np.int32
+    elif element_type == "int16" or element_type == "i16":
+        return np.int16
+    elif element_type == "int8" or element_type == "i8":
+        return np.int8
+    elif element_type == "bfloat16" or element_type == "bf16":
+        raise ValueError(
+            "Type 'bfloat16' is not supported along this path through the program (something went wrong)."
+        )
+    else:
+        raise ValueError("Invalid or unsupported element type: " + element_type)
+
+
+def write_input(bin_filename, num_elements, element_type, input_number):
+
+    # Random integer values in range [lower_bound, upper_bound)
+    # will be generated for the input data.
+    lower_bound = 0
+    upper_bound = 5
```

Comment:
This seems to be a small range for integer type.

Reply:
I've changed it to 10.